### ISSUE TITLE
Cosmetic: Buttons "End poll" and "Delete Poll" only visible for creator

### ISSUE THREAD

**Issue by Twilek-de** [matterpoll/matterpoll#73]
It would be nicer if only the poll creator would see the "End poll" and "Delete Poll" buttons. Maybe admins should have the power to end and delete polls as well...

**Comment by Hanzei** [comment]
I don't think this is possible with the current Mattermost system. The problem is that every user sees the same message and therefore the same buttons.

@kaakaa Do you have any idea how to accomplish this?

**Comment by Hanzei** [comment]
> Maybe admins should have the power to end and delete polls as well...

This is a nice idea. Would you open a new issue for this?

**Comment by Twilek-de** [comment]
Done

**Comment by kaakaa** [comment]
> @kaakaa Do you have any idea how to accomplish this?

Sorry, now I can't accomplish this.
But in terms of changing the display for each user, #28 is related to this issue. I supposed that it can be realized by making PostAttachment pluggable. But it is just an idea by now, and it will take more time to investigate and implement.

**Comment by lxm065** [comment]
i think Poll Settings can provider further customisation for them,such as delete,end,visible and so on.

**Comment by hanzei** [comment]
@lxm065 I'm not sure I'm getting your thoughts right. Do you want to add a Poll Setting for showing this buttons?

**Comment by lxm065** [comment]
> @lxm065 I'm not sure I'm getting your thoughts right. Do you want to add a Poll Setting for showing this buttons?

yeah,such like /poll "it's all right?" "yes" "no"  --progress --visible 

by the way,is there any other api command?i find that hubot can't send slash commands
https://github.com/slackapi/hubot-slack/issues/202

**Comment by hanzei** [comment]
> yeah,such like /poll "it's all right?" "yes" "no" --progress --visible


> by the way,is there any other api command?i find that hubot can't send slash commands
slackapi/hubot-slack#202

This is out of the scope of this issue. Could you please open a new one or post this question in the [Matterpoll Channel](https://community.mattermost.com/core/channels/matterpoll)?

**Comment by hanzei** [comment]
> yeah,such like /poll "it's all right?" "yes" "no" --progress --visible

I see. But why would you allow everyone to end and delete a poll? I don't see the use case. 

> by the way,is there any other api command?i find that hubot can't send slash commands
slackapi/hubot-slack#202

This is out of the scope of this issue. Could you please open a new one or post this question in the [Matterpoll Channel](https://community.mattermost.com/core/channels/matterpoll)?x

**Comment by lxm065** [comment]
> > yeah,such like /poll "it's all right?" "yes" "no" --progress --visible
> 
> I see. But why would you allow everyone to end and delete a poll? I don't see the use case.

>because i will set a hubot to do a crontab job to auto to do this poll, and then b(everyone,not creater or admin) can end and delete the poll.
only creater and admin can delete and end by default,but if you add --deleteable,it can delete or end by everyone.
add a customisation  will make it more flexible,isn't?~

> > by the way,is there any other api command?i find that hubot can't send slash commands
> > [slackapi/hubot-slack#202](https://github.com/slackapi/hubot-slack/issues/202)
> 
> This is out of the scope of this issue. Could you please open a new one or post this question in the [Matterpoll Channel](https://community.mattermost.com/core/channels/matterpoll)?

i see

**Comment by hanzei** [comment]
> because i will set a hubot to do a crontab job to auto to do this poll, and then b(everyone,not creater or admin) can end and delete the poll.
only creater and admin can delete and end by default,but if you add --deleteable,it can delete or end by everyone. Add a customisation will make it more flexible,isn't?~

This is a very special use case. I'm not 100% into adding much more Poll Settings. But please go ahead and open a new issue for this. Then we can discuss this more there.

**Comment by lxm065** [comment]
> > because i will set a hubot to do a crontab job to auto to do this poll, and then b(everyone,not creater or admin) can end and delete the poll.
> > only creater and admin can delete and end by default,but if you add --deleteable,it can delete or end by everyone. Add a customisation will make it more flexible,isn't?~
> 
> This is a very special use case. I'm not 100% into adding much more Poll Settings. But please go ahead and open a new issue for this. Then we can discuss this more there.

ok,i make it , https://github.com/matterpoll/matterpoll/issues/118

**Comment by razbag** [comment]
Hi Everyone,
Buttons "End poll" and "Delete Poll" in my mind should  only be visible for creator and admins .
I have question when this functionality should be developed , I need this plugin but wait for this future

**Comment by hanzei** [comment]
Hey @razbag,
Thanks for your interest in this project. There isn't a ETA for this feature. https://github.com/matterpoll/matterpoll/issues/28 will definitely be developed first.

Why is this feature necessary for you?

**Comment by kaakaa** [comment]
I took some effort for this issue. It would be possible to show the buttons to only the creator and system admin in **webapp**.
<img width="884" alt="スクリーンショット 2019-09-03 20 28 06" src="https://user-images.githubusercontent.com/1453749/64171262-2ea84100-ce8d-11e9-958a-601cbf891bf7.png">

However, there still be a huge problem. Since **Mattermost mobile app haven't supported plugin feature** yet, the buttons can't be hidden on mobile app. (Of course, buttons are just displayed. Users who don't have the permission can't delete/end a poll if pushing it)

@hanzei Do you think about this? Is hiding buttons on only webapp making sense?

**Comment by hanzei** [comment]
Yes. We are still "hiding" this behind the experimental setting. Thanks for working on this!

**Comment by Ansty93** [comment]
Hello!
I just got the idea of a simple/fast workaround for the delete/end poll button visiblity issue:
Instead of showing 5 buttons in the poll message, you could separate the delete/end button and make bot send a "Only for creator" message just below the poll, where the message is only visible for the creator, and the message contains delete/end poll button?
Dunno if it is any useful for you, but it might work ?

**Comment by hanzei** [comment]
> Hello!
> I just got the idea of a simple/fast workaround for the delete/end poll button visiblity issue:
> Instead of showing 5 buttons in the poll message, you could separate the delete/end button and make bot send a "Only for creator" message just below the poll, where the message is only visible for the creator, and the message contains delete/end poll button?
> Dunno if it is any useful for you, but it might probably work ?

Thanks for the suggestion @Ansty93. A "Only for creator" message is technical speaking a ephemeral message, which has a limited lifetime. Which means Matterpoll had to post this message over and over again, which is not very feasible.
